pkg/db/subscription.go

```diff
@@ -8,7 +8,11 @@ import (
 	"go.uber.org/zap"
 )
 
-type PollableDBQuery[ValueType any] func(ctx context.Context, lastSeenID int64, numRows int32) (results []ValueType, lastID int64, err error)
+type PollableDBQuery[ValueType any, CursorType any] func(
+	ctx context.Context,
+	lastSeen CursorType,
+	numRows int32,
+) (results []ValueType, nextCursor CursorType, err error)
```

Comment:
Main change is here - `PollableDBQuery` now uses a generic cursor type rather than an `int64`